genie/validate.py

```diff
@@ -1,125 +1,117 @@
-except ImportError:
-    from urllib.parse import urlparse
 import logging
 logging.basicConfig()
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
-# Validates annotations on Synapse
-# def validateAnnotations(fileList):
-#     logger.info("VALIDATING ANNOTATIONS")
-#     notcorrect = []
-#     for i,ID in enumerate(fileList['entity.id']):
-#         foo = syn.get(ID, downloadFile=False)
-#         required_annot = ["center","dataType","fileType","disease","consortium",
-#         "platform","tissueSource","organism","dataSubType"]
-#         check = [annot for annot in required_annot if foo.annotations.has_key(annot)]
-#         if len(check) != len(required_annot):
-#             notcorrect.append(fileList.iloc[i]['entity.id'])
-#     if len(notcorrect) >0:
-#         return(False)
-#     else:
-#         return(True)
-
-def synapse_login():
-    """
-    This function logs into synapse for you if credentials are saved.  
-    If not saved, then user is prompted username and password.
 
-    :returns:     Synapseclient object
-    """
-    try:
-        syn = synapseclient.login(silent=True)
-    except Exception as e:
-        logger.info("Please provide your synapse username/email and password (You will only be prompted once), or write your username and password to your bash profile under GENIE_USER and GENIE_PASS")
-        Username = os.getenv("GENIE_USER")
-        Password = os.getenv("GENIE_PASS")
-        if Username is None or Password is None:
-            Username = raw_input("Username: ")
-            Password = getpass.getpass()
-        syn = synapseclient.login(email=Username, password=Password,rememberMe=True,silent=True)
-    return(syn)
-
-def validate(syn, fileType, filePath, center, threads, oncotree_url=None, offline=False, uploadToSynapse=None, testing=False, noSymbolCheck=False):
-    """
-    This performs the validation of files
+def determine_filetype(syn, filepathlist, center, raise_error=True):
+    '''
+    Get the file type of the file by validating its filename
 
-    :returns:   Text with the errors of the chosen file
-    """
-    #CHECK: Fail if filename is incorrect
-    if not offline:
+    Args:
+        syn: Synapse object
+        filepathlist: list of filepaths to center files
+        center: Participating Center
+
+    Returns:
+        str: File type of input files
+    '''
+    filetype = None
+    # Loop through file formats
+    for file_format in PROCESS_FILES:
         try:
-            PROCESS_FILES[fileType](syn, center, threads).validateFilename(filePath)
-        except AssertionError as e:
-            raise ValueError("Your filename is incorrect!\n%s\nPlease change your filename before you run the validator again."  % e)
-    total_error, warning = PROCESS_FILES[fileType](syn, center, threads).validate(filePathList=filePath, oncotreeLink=oncotree_url, testing=testing, noSymbolCheck=noSymbolCheck)
+            validator = PROCESS_FILES[file_format](syn, center)
```

Comment:
The reason it is in a try is because i need to capture the error. But perhaps this try catch should be handled in validate.py and return the right error messages.

Reply:
just the instantiation of the `validator` should go outside of the `try` block (or in it's own). The `AssertionError` is there to catch exceptions from validating the filename, right?